frontend/packages/ux-editor/src/components/config/componentSpecificContent/AttachmentList/AttachmentListUtils.test.ts

```diff
@@ -0,0 +1,226 @@
+import {
+  dataExternalFormat,
+  reservedDataTypes,
+  dataInternalFormat,
+  getTasks,
+  selectionIsValid,
+} from './AttachmentListUtils';
+import type { LayoutSets } from 'app-shared/types/api/LayoutSetsResponse';
+import type { ApplicationMetadata } from 'app-shared/types/ApplicationMetadata';
+
+const useCasesDesc = ['current task and pdf', 'all tasks and pdf', 'current task', 'all tasks'];
+
+describe('Convert to external format: dataExternalFormat', () => {
+  describe('convert all data', () => {
+    const useCasesAllAttachments = [
+      {
+        selectedDataTypes: [
+          'attachment1',
+          'attachment2',
+          reservedDataTypes.currentTask,
+          reservedDataTypes.refDataAsPdf,
+        ],
+        availableAttachments: ['attachment1', 'attachment2'],
+        expected: [reservedDataTypes.includeAll, reservedDataTypes.currentTask],
+      },
+      {
+        selectedDataTypes: ['attachment1', 'attachment2', reservedDataTypes.refDataAsPdf],
+        availableAttachments: ['attachment1', 'attachment2'],
+        expected: [reservedDataTypes.includeAll],
+      },
+      {
+        selectedDataTypes: ['attachment1', 'attachment2', reservedDataTypes.currentTask],
+        availableAttachments: ['attachment1', 'attachment2'],
+        expected: [reservedDataTypes.currentTask],
+      },
+      {
+        selectedDataTypes: ['attachment1', 'attachment2'],
+        availableAttachments: ['attachment1', 'attachment2'],
+        expected: [],
+      },
+    ];
+
+    useCasesAllAttachments.forEach((useCase, index) => {
+      it(`should convert to external format with all attachments and ${useCasesDesc[index]}`, () => {
+        expect(dataExternalFormat(useCase.selectedDataTypes, useCase.availableAttachments)).toEqual(
+          useCase.expected,
+        );
+      });
+    });
```

Comment:
Det er lurt å sette opp en liste over forskjellige scenarier på denne måten, men det hadde vært mer oversiktlig om beskrivelsene sto sammen med oppsettet. Dette er også sårbart for endringer i rekkefølge siden indeks er eneste referanse.

Prøv gjerne å sette det opp på denne måten i stedet: https://github.com/Altinn/altinn-studio/blob/main/frontend/packages/schema-editor/src/components/NodePanel/HeadingRow/HeadingRow.test.tsx#L113
Her er beskrivelsene sammen med oppsettet for det enkelte scenariet, som også har en egen, intern type. Her er også `describe.each` brukt til å iterere over scenariene - du kan bruke `it.each` på samme måte.